file_paths.md

```diff
@@ -106,16 +115,144 @@ puppet-agent
         mcollective.pid
         mcollectived.pid
 
+<a id="windows">puppet-agent (windows)</a>
+------
+
+On recent versions of Windows, e.g. 2008 & 2012, the installation path
+defaults to `C:\Program Files\Puppet Labs` and the common app data
+directory defaults to `C:\ProgramData`. On 2003, common app data is
+under `C:\Documents and Settings\All Users\Application Data`. Also
+when installing puppet-agent 32-bit on 64-bit windows, the
+installation path defaults to `C:\Program Files (x86)\Puppet
+Labs`. The examples below assume 2008/2012 and puppet-agent (64-bit).
+
+    Path                                      Setting                        3.x
+    C:\ProgramData                                                           n/a
+
+    C:\ProgramData\PuppetLabs\code            # :codedir                     contents moved from :confdir
+        environments                          # :environmentpath
+        hiera.yaml                            # :hiera_config
+        hieradata                             # n/a
+        modules                               # user modulepath
+
+    C:\ProgramData\PuppetLabs\mcollective\etc                                same
+        client.cfg
+        facts.yaml
+        server.cfg
+
+    C:\ProgramData\PuppetLabs\mcollective\var                                same
+        log
+            mcollective.log
+
+    C:\ProgramData\PuppetLabs\puppet\etc      # :confdir                     same
+        csr_attributes.yaml                   # :csr_attributes
+        custom_trusted_oid_mapping.yaml       # :trusted_oid_mapping_file
+        puppet.conf                           # :config
+        ssl                                   # :ssldir
+
+    C:\ProgramData\PuppetLabs\puppet\cache    # :vardir                      C:\ProgramData\PuppetLabs\puppet\var
+        bucket                                # :bucketdir
+        client_yaml                           # :clientyamldir
+        client_data                           # :client_datadir
+        clientbucket                          # :clientbucketdir
+        devicedir                             # :devices
+        facts.d                               # :pluginfactdest (pluginsync'ed)
+        lib                                   # :libdir
+        facts                                 # used to generate :factpath
+        puppet-module                         # :module_working_dir
+        reports                               # :reportdir
+        server_datadir                        # :server_data
+        state                                 # :statedir
+        yaml                                  # :yamldir
+
+    C:\ProgramData\PuppetLabs\facter                                         same
```

Comment:
For facter, https://github.com/puppetlabs/facter/pull/855, and ticket filed for cfacter: https://tickets.puppetlabs.com/browse/CFACT-228